### DISCUSSION TITLE
Parsing Go

### DISCUSSION THREAD

**Discussion by leaanthony** [alecthomas/participle#188] [Q&A]
Seems like an obvious use case, but can Participle easily parse Go? I'm assuming not as there's no examples?

**Comment by alecthomas** [comment]
I believe it could, but I haven't made or seen an attempt to write a Go grammar for Participle. What is your use case? I assume you know about [go/parser](https://pkg.go.dev/go/parser)?

**Reply by leaanthony** [reply]
Yeah, Irina the default parser really hard to use, especially when dealing with external packages. My use case is I need to trace back from a variable to its struct type, parsethe definition of the struct for the method signatures.
